pulsar-client-kafka-compat/pulsar-client-kafka_0_8/src/main/java/org/apache/kafka/clients/consumer/ConsumerConnector.java

```diff
@@ -0,0 +1,230 @@
+    public List<PulsarKafkaStream<byte[], byte[]>> createMessageStreamsByFilter(TopicFilter topicFilter) {
+        throw new UnsupportedOperationException("method not supported");
+    }
+
+    public List<PulsarKafkaStream<byte[], byte[]>> createMessageStreamsByFilter(TopicFilter topicFilter, int arg1) {
+        throw new UnsupportedOperationException("method not supported");
+    }
+
+    public <K, V> List<PulsarKafkaStream<K, V>> createMessageStreamsByFilter(TopicFilter topicFilter, int arg1,
+            Decoder<K> keyDecoder, Decoder<V> valueDecoder) {
+        throw new UnsupportedOperationException("method not supported");
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<CompletableFuture<Void>> commitOffsetsAsync() {
+        return topicStreams.stream().map(stream -> (CompletableFuture<Void>) stream.commitOffsets())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Commit the offsets of all broker partitions connected by this connector.
+     */
+    public void commitOffsets() {
+        commitOffsetsAsync();
+    }
+
+    public void commitOffsets(boolean retryOnFailure) {
+        FutureUtil.waitForAll(commitOffsetsAsync()).handle((res, ex) -> {
+            if (ex != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Failed to commit offset {}, retrying {}", ex.getMessage(), retryOnFailure);
+                }
+                if (retryOnFailure) {
+                    this.executor.schedule(() -> commitOffsets(retryOnFailure), 30, TimeUnit.SECONDS);
+                }
+            }
+            return null;
+        });
+    }
+
+    /**
+     * Shut down the connector
+     */
+    public void shutdown() {
+        if (executor != null) {
+            executor.shutdown();
+        }
+        if (topicStreams != null) {
+            topicStreams.forEach(stream -> {
+                try {
+                    stream.close();
+                } catch (Exception e) {
+                    log.warn("Failed to close stream {}, {}", stream, e.getMessage());
+                }
+            });
+        }
+        try {
+            client.close();
+        } catch (PulsarClientException e) {
+            log.warn("Failed to close client {}", e.getMessage());
+        }
+    }
+
+    private void resetOffsets(Consumer<byte[]> consumer, SubscriptionInitialPosition strategy) {
+        if (strategy == null) {
+            return;
+        }
+        log.info("Resetting partition {} for group-id {} and seeking to {} position", consumer.getTopic(),
+                consumer.getSubscription(), strategy);
+        try {
+            if (strategy == SubscriptionInitialPosition.Earliest) {
+                consumer.seek(MessageId.earliest);
+            } else {
+                consumer.seek(MessageId.latest);
+            }
+        } catch (PulsarClientException e) {
+            log.warn("Failed to reset offeset for consumer {} to {}, {}", consumer.getTopic(), strategy,
```

Comment:
Please fix the typo in the word `offset`. 
Maybe, it makes sense to provide the whole exception as the last parameter (and remove it from the parametrized string) instead of the exception message.